### PR TITLE
 DeveloperGuide: add solution to src being treated as normal folder

### PR DESCRIPTION
**Pull request by eugenepeh**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -66,3 +67,7 @@ OSes. Convert the `actual.txt` to the format used by your OS using some https://
 * Problem: Test fails during the very first time. +
 Solution: The output of the very first test run could be slightly different because the program
 creates a new storage file. Tests should pass from the 2nd run onwards.
+
+* Problem: Marked directory, `Sources Root`, keep getting unmarked. +
+Solution: This can happen if you've accidentally import the gradle library. +
```
Comment (yamgent) [inline docs/DeveloperGuide.adoc]: ...accidentally ~~import~~ imported...

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -66,3 +67,7 @@ OSes. Convert the `actual.txt` to the format used by your OS using some https://
...
+Please try to disable `"Use auto-import"` in `Settings (Preferences)` > `Build...` > `Build Tools` > `Gradle`.
```
Comment (yamgent) [inline docs/DeveloperGuide.adoc]: At the end, remind the student to re-mark the `src/` as "Source Root" after disabling Gradle.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -66,3 +67,7 @@ OSes. Convert the `actual.txt` to the format used by your OS using some https://
 * Problem: Test fails during the very first time. +
 Solution: The output of the very first test run could be slightly different because the program
 creates a new storage file. Tests should pass from the 2nd run onwards.
+
+* Problem: Marked directory, `Sources Root`, keep getting unmarked. +
```
Comment (yamgent) [inline docs/DeveloperGuide.adoc]: Problem: `AddressBook` cannot be run, and `src/` is treated as a normal folder instead of as a `Sources Root`.

**Comment by eugenepeh** [comment]
>Perhaps during step 4 ("Locate the project directory and click OK"), the student might have accidentally selected build.gradle instead of the "project directory"? Selecting build.gradle will definitely make IntelliJ import the Gradle project.

Hmm... I went through the set up instructions with the student to make sure he follow the developer guide. His gradle always perform sync on launch and disabling the wrapper stops it somehow. I was not able to replicate it, I thought it was probably because that was his only gradle distribution. 

I guess I will just add the troubleshooting section for now.

**Comment by eugenepeh** [comment]
@yamgent 

As pointed out by another student who also had this problem, the original instruction to uncheck gradle-wrapper during the set up seems to solve his problem.

Mind if I add it back to the Developer Guide?

**Comment by damithc** [comment]
Note that gradle is for se-edu devs. We don't want students to use gradle at AB-1 stage as the topic is not covered so early in the semester.

**Comment by yamgent** [comment]
> @yamgent
> 
> As pointed out by another student who also had this problem in nusCS2113-AY1819S1/forum#4 (comment), the original instruction to uncheck gradle-wrapper during the set up seems to solve his problem as well.
> 
> Mind if I add it back to the Developer Guide?

If it really solves it for a few people, I don't mind adding it back in.

**Comment by pyokagan** [comment]
@damithc 
> We don't want students to use gradle at AB-1 stage as the topic is not covered so early in the semester.

I'm not suggesting that, I'm suggesting making it such that the `build.gradle` config just *happens* to contain the same configuration as what IntelliJ should ideally figure out by itself if the "use existing project sources" import option is used.

Note that there is precedent for this: [AB-2 has the same setup](https://github.com/se-edu/addressbook-level2/blob/0a1dd3326ab6202bb50ab99bf5e7fbbb00bd559e/build.gradle#L19-L30), even though students are still not using Gradle at that stage. (Notice that there were not similar complaints from students in previous semesters)

**Comment by yamgent** [comment]
> If we just add the java plugin to the build.gradle file (along with all the necessary configuration), would it help resolve the problem without needing any additional "troubleshooting" step?

Would definitely help, students won't have to do additional stuff this way. I am leaning towards this suggestion, especially as you mentioned that AB-2 has the same setup.

**Comment by eugenepeh** [comment]
>Would definitely help, students won't have to do additional stuff this way. I am leaning towards this suggestion, especially as you mentioned that AB-2 has the same setup.

Agree, sounds good. Didn't realize AB2 has already implemented this.

**Pull request merged**